### PR TITLE
[Inductor] Preserve positional args in pre-grad batch math fusion

### PR DESCRIPTION
**Pull request by warrendeng**
Summary:
BatchMathOpsPreGradFusion currently groups only keyword arguments, and ignores positional arguments. For models that use positional arguments for ops like torch.clamp and torch.nan_to_num, this can lead to dropped parameters/errors. An error we recently encountered on an internal model is 
`BackendCompilerFailed: backend='inductor' raised:\nValueError: clamp called but both min and max are none!` due to positional min and max arguments of torch.clamp being dropped

In this diff we include positional arguments in both the fusion group key and the batched call. Add regression coverage for positional clamp bounds and distinct bound groups.

This allows us to use positional arguments for the following 4 ops:
* torch.detach(input) (no change)
* torch.clamp(input, min, max)
*  torch.nan_to_num(input, nan, posinf, neginf)
* torch.nn.functional.dropout(input, p, training, inplace)

Test Plan:
- buck2 test fbcode//mode/dev-nosan fbcode//caffe2/test/inductor:group_batch_fusion -- test_math_op_fusion_predispatch_positional_args
- buck2 test fbcode//mode/dev-nosan fbcode//caffe2/test/inductor:group_batch_fusion -- test_math_op_fusion
- arc lint --never-apply-patches fbcode/caffe2/torch/_inductor/fx_passes/group_batch_fusion.py fbcode/caffe2/test/inductor/test_group_batch_fusion.py

Differential Revision: D113485896

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Review by jananisriram — APPROVED** [review]
lgtm, thanks! can we confirm that the failing CI is unrelated?

**Comment by warrendeng** [comment]
@claude please review the CI failures and diagnose if they were caused by this commit

**Review by han3101 — APPROVED** [review]
